src/main/java/core/basesyntax/ArrayList.java

```diff
@@ -1,48 +1,119 @@
 package core.basesyntax;
 
+import java.util.NoSuchElementException;
+
 public class ArrayList<T> implements List<T> {
+    private static final double CAPACITY_INCREACE_VALUE = 1.5;
+    private static final int INITIAL_CAPACITY = 10;
+    private T [] storage;
+    private int size;
+
+    public ArrayList() {
+        storage = (T[]) new Object[INITIAL_CAPACITY];
+    }
+
     @Override
     public void add(T value) {
-
+        if (size == storage.length) {
+            increaseCapacity((int)(size * CAPACITY_INCREACE_VALUE));
+        }
+        storage[size] = value;
+        size++;
     }
 
     @Override
     public void add(T value, int index) {
-
+        if (index < 0 || index > size) {
+            throw new ArrayListIndexOutOfBoundsException("Index is out of bound!");
+        }
+        if (size == storage.length) {
+            increaseCapacity((int)(size * CAPACITY_INCREACE_VALUE));
+        }
+        System.arraycopy(storage, index, storage, index + 1, size - index);
+        storage[index] = value;
+        size++;
     }
 
     @Override
     public void addAll(List<T> list) {
-
+        int neededCapacity = size + list.size();
+        if (neededCapacity > storage.length) {
+            increaseCapacity(neededCapacity);
+        }
+        for (int i = 0; i < list.size(); i++) {
+            storage[size + i] = list.get(i);
+        }
+        size += list.size();
```

Comment:
too complex logic, go through all list and call add method that you implemented above